### PR TITLE
[AEA-1185] Feature/libp2p PoR ledger id support

### PR DESCRIPTION
**Pull request by lrahmani**
## Proposed changes

This PR adds:
- ledger_id support for proof of representation/agent record
- ethereum support for ACN
- fixes associated tests

### CODE REVIEW

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go [hunk 1]
```diff
@@ -231,6 +245,12 @@ func BTCPubKeyFromFetchAIPublicKey(publicKey string) (*btcec.PublicKey, error) {
 	return pbk, err
 }
 
+// BTCPubKeyFromEthereumPublicKey create libp2p public key from ethereum uncompressed
+//  hex encoded secp256k1 key
+func BTCPubKeyFromEthereumPublicKey(publicKey string) (*btcec.PublicKey, error) {
+	return BTCPubKeyFromUncompressedHex(publicKey[2:])
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/utils/utils.go:251]: so dropping `0x` I imagine? Worth adding as a comment probably.

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go [hunk 2]
```diff
@@ -277,6 +297,29 @@ func ParseFetchAISignature(signature string) (*btcec.Signature, error) {
 	return sigBTC, err
 }
 
+// ParseFetchAISignature create btcec Signature from base64 formated, string (not DER) encoded RFC6979 signature
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/utils/utils.go:300]: comment typo

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go [hunk 3]
```diff
@@ -337,6 +380,30 @@ func SignFetchAI(message []byte, privKey string) (string, error) {
 	return encodedSignature, nil
 }
 
+// VerifEthereumSignatureBTC verify the RFC6967 string-encoded signature of message using FetchAI public key
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/utils/utils.go:383]: typo

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go [hunk 4]
```diff
@@ -378,10 +464,53 @@ func FetchAIAddressFromPublicKey(publicKey string) (string, error) {
...
 	return addr, err
 }
 
+// EthereumAddressFromPublicKey get wallet address from hex encoded secp256k1 public key
+// format from: https://github.com/fetchai/agents-aea/blob/master/aea/crypto/ethereum.py#L330
+// check also: https://github.com/ethereum/go-ethereum/blob/master/crypto/crypto.go#L263
+func EthereumAddressFromPublicKey(publicKey string) (string, error) {
+	var addr string
+	var err error
+	hexBytes, err := hex.DecodeString(publicKey[2:])
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/utils/utils.go:477]: see above, worth adding in comment that `0x` is dropped

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go [hunk 5]
```diff
@@ -378,10 +464,53 @@ func FetchAIAddressFromPublicKey(publicKey string) (string, error) {
...
+	if err != nil {
+		return addr, err
+	}
+	hash := sha3.NewLegacyKeccak256()
+	_, err = hash.Write(hexBytes)
+	if err != nil {
+		return addr, err
+	}
+	sha3KeccakHash := hash.Sum(nil)
+	return encodeChecksumEIP55(sha3KeccakHash[12:]), nil
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/p2p_libp2p/utils/utils.go:487]: what about the first 12? worth adding a comment what's going on

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go (hunk 1 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/utils/utils.go:254]: I don't think we need it, should be transparent

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go (hunk 4 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/utils/utils.go:490]: same, I believe it should be transparent, as dropping `0x` is related to hex decoder in use

File: packages/fetchai/connections/p2p_libp2p/utils/utils.go (hunk 5 above)
Comment (lrahmani) [reply packages/fetchai/connections/p2p_libp2p/utils/utils.go:500]: this is just how ethereum addresses are computed.
see function header comments

File: tests/conftest.py [hunk 6]
```diff
@@ -634,7 +634,7 @@ def update_default_ethereum_ledger_api(ethereum_testnet_config):
 @pytest.mark.integration
 @pytest.mark.ledger
 @pytest.fixture(scope="session")
-@action_for_platform("Linux", skip=False)
+# @action_for_platform("Linux", skip=False)
```
Comment (DavidMinarsch) [inline tests/conftest.py:637]: revert!

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by marcofavorito — APPROVED** [review]
LGTM

**Pull request merged**